Integrations/EclecticIQ/EclecticIQ.py

```diff
@@ -0,0 +1,640 @@
+
+    maliciousness_dictionary = {
+        'unknown': 0,
+        'safe': 1,
+        'low': 2,
+        'medium': 2,
+        'high': 3
+    }
+
+    for i in maliciousness_list[maliciousness_list.index(threshold):]:
+        maliciousness_dictionary[i] = 3
+
+    return maliciousness_dictionary[maliciousness]
+
+
+''' COMMANDS + REQUESTS FUNCTIONS '''
+
+
+def test_module():
+    """
+
+    The function which runs when clicking on Test in integration settings
+
+
+    Returns
+    -------
+    str
+        ok if getting observable successfully
+
+    """
+
+    get_observable('8.8.8.8')
+    demisto.results('ok')
+
+
+def login():
+    """
+
+    Logins to EclecticIQ API with given credentials and sets the returned token in the headers
+
+    """
+
+    cmd_url = '/api/auth'
+    cmd_json = {
+        'password': PASSWORD,
+        'username': USERNAME
+    }
+    response = http_request('POST', cmd_url, cmd_json=cmd_json)
+    if 'token' in response:
+        token = response['token']
+    else:
+        return_error('Failed to retrieve token')
+    HEADERS['Authorization'] = 'Bearer {}'.format(token)
+
+
+def ip_command():
+    """
+
+    Gets reputation of an EclecticIQ IPv4 observable
+
+    Parameters
+    ----------
+    ip : str
+        IPv4 to get reputation of
+
+    Returns
+    -------
+    entry
+        Translated DBot Score
+
+    """
+
+    ip = demisto.args()['ip']
+
+    response = get_observable(ip)
+
+    if 'total_count' in response and response['total_count'] == 0:
+        human_readable = 'No results found'
+
+    integration_outputs = []
+    standard_ip_outputs = []
+
+    observables = response.get('data')
+
+    score = 0
+
+    for observable in observables:
+        meta = observable.get('meta', {})
+        maliciousness = meta.get('maliciousness')
+        score = maliciousness_to_dbotscore(maliciousness, IP_THRESHOLD)
+
+        integration_outputs.append({
+            'Address': ip,
+            'Created': observable.get('created_at'),
+            'LastUpdated': observable.get('last_updated_at'),
+            'ID': observable.get('id'),
+            'Maliciousness': maliciousness
+        })
+
+        standard_ip_output = {
+            'Address': ip
+        }
+        if score == 3:
+            standard_ip_output['Malicious'] = {
+                'Vendor': 'EclectiqIQ',
+                'Description': 'EclectiqIQ maliciousness confidence level: ' + maliciousness
+            }
+
+        standard_ip_outputs.append(standard_ip_output)
+
+    dbot_output = {
+        'Type': 'ip',
+        'Indicator': ip,
+        'Vendor': 'EclecticIQ',
+        'Score': score
+    }
+
+    context = {
+        'DBotScore': dbot_output
+    }
+
+    if observables:
+        human_readable_title = 'EclecticIQ IP reputation - {}'.format(ip)
+        human_readable = tableToMarkdown(human_readable_title, integration_outputs)
+        context['EclecticIQ.IP(val.ID===obj.ID)'] = integration_outputs
```

Comment:
directly add this to context and use CreateContent with removeNulls=True to avoid empty fields.
the human_readable can also be used directly in the entry (it will write "no result")

Reply:
what do you mean by "directly" ?

Reply:
relates to previous comment about the markdown